usr/src/lib/brand/bhyve/zone/boot.c

```diff
@@ -262,6 +263,8 @@ add_disk(char *disk, char *path, char *slotconf, size_t slotconf_len)
 
 	if (is_env_string("device", disk, "model", "virtio")) {
 		model = "virtio-blk";
+		if (is_env_string("device", disk, "nodelete", "true"))
+			nodelstr = ",nodelete";
```

Comment:
Do you still want to call it nodelete here if the vmadm site is going to go with notrim/nounmap?

Reply:
notrim or nounmap sounds corrected that nodelete, nodelete may imply a WORM type device for some people.

Reply:
The blockif feature is called delete (from upstream), so either way there's going to be a discrepancy somewhere. Since people generally aren't running bhyve directly, it seemed ok to do that here (using two different terms in the code feels like it'd be more confusing) though it would show up in pargs(1M). those that are running bhyve directly should hopefully not get confused (since the assumption is if they're attempting that, they should already have some sort of advanced knowledge).

Reply:
We could maybe change the attribute in the zone configuration -- that at least wouldn't mix things up too much.

Reply:
I think that makes sense.  The device string from bhyve's point of view should be consistent. 
We should make the zonecfg property match what vmadm is using though.  Thanks, I am happy with this.